### ISSUE TITLE
Intermittent crash in /css-transforms-1_dev/html/svg-rotate-angle-90-014.htm and /css21_dev/html4/border-right-018.htm and /css-transforms-1_dev/html/transofrmed-rotateY-1.htm

### ISSUE THREAD

**Issue by Manishearth** [servo/servo#12553]
cc @asajeffrey looks like things are out of order during shutdown?

```
Tests with unexpected results:
  ▶ CRASH [expected PASS] /css-transforms-1_dev/html/svg-rotate-angle-90-014.htm
  │ 
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ ERROR:servo: called `Result::unwrap()` on an `Err` value: RecvError
  └ thread panicked while processing panic. aborting.
```

**Comment by jdm** [comment]
Same symptom as #12540.

**Comment by jdm** [comment]
```
  ▶ CRASH [expected PASS] /css21_dev/html4/border-right-018.htm
  │ 
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ ERROR:servo: called `Result::unwrap()` on an `Err` value: RecvError
  └ thread panicked while processing panic. aborting.
```

**Comment by jdm** [comment]
```
  ▶ CRASH [expected PASS] /css-transforms-1_dev/html/transofrmed-rotateY-1.htm
  │ 
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ ERROR:servo: called `Result::unwrap()` on an `Err` value: RecvError
  └ thread panicked while processing panic. aborting.
```

**Comment by asajeffrey** [comment]
Oh good. Double panic.

**Comment by asajeffrey** [comment]
IRC chat with @jdm: http://logs.glob.uno/?c=mozilla%23servo&s=27+Jul+2016&e=27+Jul+2016#c489397

TL;DR: mutex.lock() can panic https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.lock, which may be the source of the double panic. Removing the unnecessary mutex should fix the double panic.

**Comment by jdm** [comment]
These tests no longer panic, and this issue doesn't match any tests that we run.
